### PR TITLE
Update carrierwave to lastest

### PR DESCRIPTION
**Pull request by erluti**
Updates carrierwave gem to the latest.

`spec/uploaders/evss_claim_document_uploader_spec.rb` had a spec that was failing.  After investigating it was never working as intended, so it was removed.  Instead a spec for `ValidateFileSize` was added since it had no tests.

### CODE REVIEW

**Comment by annaswims** [comment]
`rspec ./spec/uploaders/evss_claim_document_uploader_spec.rb:126 # EVSSClaimDocumentUploader#store! raises an error when the file is larger than 25 megabytes`

was a poorly written test. it was expecting an error, due to file size but instead the error raised was 
`CarrierWave::IntegrityError: You are not allowed to upload nil files, allowed types: pdf, gif, tiff, tif, jpeg, jpg, bmp, txt` which is a kind of `CarrierWave::UploadError`

**Comment by annaswims** [comment]
There are breaking changes noted in the changelog.  Like - `extension_white_list ~> extension_whitelist`

https://github.com/carrierwaveuploader/carrierwave/blob/3356634d80d372c458030460978c3b63c006e042/CHANGELOG.md

**Comment by annaswims** [comment]
I'm trying to standardize the error messages of file uploads in #5022 and added some tests that cover file upload size [here](https://github.com/department-of-veterans-affairs/vets-api/pull/5059/files#diff-a03e7121868faa25fc4afae420bff0eeR348) and

**Comment by erluti** [comment]
> There are breaking changes noted in the changelog. Like - `extension_white_list ~> extension_whitelist`
> 
> https://github.com/carrierwaveuploader/carrierwave/blob/3356634d80d372c458030460978c3b63c006e042/CHANGELOG.md

didn't realize that one applied!  thanks!

**Comment by erluti** [comment]
> I'm trying to standardize the error messages of file uploads in #5022 and added some tests that cover file upload size [here](https://github.com/department-of-veterans-affairs/vets-api/pull/5059/files#diff-a03e7121868faa25fc4afae420bff0eeR348) and

are you suggesting I try to copy some here or do you want to finish that PR before the upgrade is merged?

**Comment by annaswims** [comment]
I was thinking I'd merge# 5059 first, since it adds a bit of coverage. It's good to go as soon as it gets an approval, I think.

**Comment by annaswims** [comment]
(sorry, deleted my previous incorrect comment right after I wrote it.) 

I just wanted you to be aware.  I'm going to hold off merging mine until I've tested with the FE.

File: spec/uploaders/validate_file_size_spec.rb
Comment (LindseySaari) [inline spec/uploaders/validate_file_size_spec.rb:9]: To make this more uniform with Annas PR, can you make the MAX_FILE_SIZE a class method instead of a constant. See [here](https://github.com/department-of-veterans-affairs/vets-api/pull/5059/files#diff-5e5224c9764baaff6de004f124c5434cL10)

Comment (erluti) [reply spec/uploaders/validate_file_size_spec.rb:9]: I feel like that's out of scope for this change.  It might even make sense as it's own PR

**Review by LindseySaari — APPROVED** [review]

File: spec/uploaders/validate_file_size_spec.rb
Comment (annaswims) [reply spec/uploaders/validate_file_size_spec.rb:9]: So.... I think the solution is to get rid of the ValidateFileSize module altogether.  Carrierwave has [a size range method](https://github.com/carrierwaveuploader/carrierwave/blob/f15f4ba71dd2c78e8df494bf65938bbaf13960f9/lib/carrierwave/uploader/file_size.rb#L21)

Comment (LindseySaari) [reply spec/uploaders/validate_file_size_spec.rb:9]: Oh, good thought Anna. We can probably break this work out into a separate PR

Comment (annaswims) [reply spec/uploaders/validate_file_size_spec.rb:9]: Good call! FWIW, I ran the specs for our two branches together and this was the only test to fail.  I can fix the conflict after you merge.

Comment (erluti) [reply spec/uploaders/validate_file_size_spec.rb:9]: I'll go forward with the merge here, then.

**Pull request merged**
